app/daemon.go

```diff
@@ -244,8 +253,54 @@ func startManager(c *cli.Context) error {
 	return nil
 }
 
-func environmentCheck() error {
+func environmentCheck(kubeconfigPath, currentNodeID string) error {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return errors.Wrap(err, "unable to get client config")
+	}
+
+	kubeClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "unable to get k8s client")
+	}
+
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		logrus.Warnf("Cannot detect pod namespace, environment variable %v is missing, using default namespace", types.EnvPodNamespace)
+		namespace = corev1.NamespaceDefault
+	}
+
+	// Check if mount propagation is supported
+	if err := checkMountPropagation(kubeClient, namespace, currentNodeID); err != nil {
+		return err
+	}
+
+	kubeNode, err := kubeClient.CoreV1().Nodes().Get(context.Background(), currentNodeID, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get node when checking kernel version")
+	}
+
+	// Check if kernel version is recommended
+	if err := checkKernelVersion(kubeNode, currentNodeID); err != nil {
```

Comment:
`kubeNode` already contains the `nodeID`. Can we remove the parameter `currentNodeID`?

```suggestion
	if err := checkKernelVersion(kubeNode); err != nil {
```

Reply:
Modified. Thanks.